src/aframe-components/scene-timer.js

```diff
@@ -0,0 +1,277 @@
+/* global AFRAME, STREET */
+
+/**
+ * Scene Timer Component
+ * A global timer component that manages time for the entire scene.
+ * Provides play, pause, stop functionality and timing information.
+ */
+AFRAME.registerComponent('scene-timer', {
+  schema: {
+    autoStart: { type: 'boolean', default: false },
+    format: { type: 'string', default: 'mm:ss:ff' } // mm:ss:ff, ss:ms, or raw
+  },
+
+  init: function () {
+    // Initialize timer state
+    this.elapsedTime = 0; // Time in milliseconds
+    this.startTime = null;
+    this.timerActive = false;
+    this.isPaused = false;
+    this.frameRate = 30; // Assumed frame rate for frame count
+
+    // Bind event handlers once to reuse the same function references
+    this.boundStartTimer = this.startTimer.bind(this);
+    this.boundPauseTimer = this.pauseTimer.bind(this);
+    this.boundStopTimer = this.stopTimer.bind(this);
+    this.boundReset = this.reset.bind(this);
+    this.boundSetTimeHandler = (event) => {
+      if (event.detail) {
+        // Handle time setting
+        if (typeof event.detail.time === 'number') {
+          this.setTime(event.detail.time);
+        }
+
+        // Handle format changes
+        if (typeof event.detail.format === 'string') {
+          this.el.setAttribute('scene-timer', 'format', event.detail.format);
+          console.log('Scene timer format changed to', event.detail.format);
+        }
+      }
+    };
+
+    // Register timer event handlers
+    this.bindEvents();
+
+    // Auto-start if configured
+    if (this.data.autoStart) {
+      this.play();
+    }
+
+    // Register this component in the global STREET object if available
+    if (typeof STREET !== 'undefined') {
+      if (!STREET.timer) {
+        STREET.timer = {};
+      }
```

Comment:
All the code related to setting a global STREET.timer and also setting it on a react ref I really don't like it, I find it fragile. I did similar code in a previous project and I had race condition between the defined global and the react code that wanted to access it.
What I would do if I need it in a component is just use `this.el.sceneEl.components['scene-timer']` and if you need it in react, create an easy getTimerComponent function that does
```js
function getTimerComponent() {
  return document.querySelector('a-scene')?.components['scene-timer'];
}
```

Reply:
Thanks for the push, you are right, I removed the use of a global to access the timer
1eec09617176a312a0a1928fa0ea7bd0426224f7